spec/factories/course_assessment_submissions.rb

```diff
@@ -6,8 +6,14 @@
     transient do
       grader { User.stamper }
       auto_grade true # Used only with any of the submitted or finalised traits.
+      user { creator }
```

Comment:
Actually this transient attribute is used for the `experience_points_record` factory, but since there is a relationship between the `user` (in exp_record) and the `creator` (in submissions), this has to exist unless I fix experience points record factory (in another PR).

Arising from this I think it will be fair if all factories are defined as `create(:submission, assessment, creator: user)` and not `create(:submission, assessment, user: user)`.